File: Android/client/lib/api/insights.dart
import 'api_client.dart';

class InsightsApi {
  static final _api = BizOsApiClient.instance;

  static Future<Response<dynamic>> today() =>
      _api.get('/client/insights/today');
  static Future<Response<dynamic>> range({Map<String, dynamic>? query}) =>
      _api.get('/client/insights/range', query: query);
  static Future<Response<dynamic>> stockAlerts() =>
      _api.get('/client/insights/stock-alerts');
  static Future<Response<dynamic>> chat(Map<String, dynamic> data) =>
      _api.post('/client/insights/chat', data: data);
}

File: Desktop/src/api/insights.ts
import { api } from './axios';
import type { DailyMetric, StockAlert } from '@/types/insight';

export interface InsightTodayResponse {
  latestMetric: DailyMetric | null;
  lowStock: StockAlert[];
}

export const insightApi = {
  today: () =>
    api
      .get<{ data: InsightTodayResponse }>('/client/insights/today')
      .then((r) => r.data.data),

  range: (params: { from?: string; to?: string; period?: string } = {}) =>
    api
      .get<{ data: DailyMetric[] }>('/client/insights/range', { params })
      .then((r) => r.data.data),

  chat: (text: string) =>
    api
      .post<{ data: { reply: string; tokensUsed: number } }>(
        '/client/insights/chat',
        { text }
      )
      .then((r) => r.data.data),

  stockAlerts: () =>
    api
      .get<{ data: StockAlert[] }>('/client/insights/stock-alerts')
      .then((r) => r.data.data),
};